runtime/debug/outer-port-attachment.js

```diff
@@ -37,11 +37,13 @@ export class OuterPortAttachment {
     }
   }
 
-  onSynchronize({handle, target, callback, modelCallback, type, particleId}) {
-    this._callbackRegistry[callback] = this._describeHandleCall(
-      {operation: `on-${type}`, handle, particleId});
-    this._callbackRegistry[modelCallback] = this._describeHandleCall(
-      {operation: 'sync-model', handle, particleId});
+  // TODO: particle IDs for proxy calls?
```

Comment:
If we don't plug in IDs it's impossible for devtools to associate these events with particles. Is this something we could add? particleId is currently expected in _describeHandleCall so this will fail I'm afraid.

Reply:
The issue is that each StorageProxy can have multiple observer particles but only fires the API channel calls once. Dealing with multiple IDs probably means adding a call specifically for the devtools to associate the extra ones with the same store.

There's another small wrinkle: SynchronizeProxy is also called when the proxy becomes desynchronised from the backing store. I think it would be fine for the devtools to just ignore those calls - the association between particle(s) and stores has already been established, this extra call is just to resync the data - so the OuterPortAttachment implementation just needs to know that this is a desync-specific call. And that can be done quite easily by not passing any particle IDs through.

Reply:
Mick, Doug & I talked through this, happy to chat again with you too @piotrswigon ; however our conclusion was:

The new API breaks several assumptions made by DevTools about data flow, including:
* data flow events are not always specific to particles any more
* data fetch calls do not always result in messages on the channel

This means that the DevTools data flow panel needs a fairly extensive overhaul in order to be able to work against the new UI (just adding particle IDs to the new calls won't work).

Our judgement at the moment is that switching to the new API is more important than maintaining the data flow view, because the new API is fixing bugs people are currently experiencing. This does mean that one of us will need to go and fix DevTools at some point in the future, though.

Are you OK with this line of reasoning?

Reply:
Yeah, definitely, new API is much more important, go ahead.

Once we'll need the tool like dataflow we can take a look how to visualize those calls and overhaul that panel.